libethcore/Farm.h

```diff
@@ -223,37 +223,36 @@ class Farm: public FarmFace
 		return m_isMining;
 	}
 
-	/**
-	 * @brief Get information on the progress of mining this work package.
-	 * @return The progress with mining so far.
-	 */
-	WorkingProgress const& miningProgress(bool hwmon = false) const
-	{
-		WorkingProgress p;
-		p.ms = 0;
-		p.hashes = 0;
-		{
-			Guard l2(x_minerWork);
-			for (auto const& i : m_miners) {
-				p.minersHashes.push_back(0);
-				if (hwmon)
-					p.minerMonitors.push_back(i->hwmon());
-			}
-		}
-
-		for (auto const& cp : m_lastProgresses) {
-			p.ms += cp.ms;
-			p.hashes += cp.hashes;
-			for (unsigned int i = 0; i < cp.minersHashes.size(); i++)
-			{
-				p.minersHashes.at(i) += cp.minersHashes.at(i);
-			}
-		}
-
-		Guard l(x_progress);
```

Comment:
This mutex was only for writers, but reader were not protected. With the extended scope of the `x_minerWork` it is not needed any more.